posydon/binary_evol/flow_chart.py

```diff
@@ -388,6 +402,9 @@ def initial_eccentricity_flow_chart(FLOW_CHART=None, CHANGE_FLOW_CHART=None):
         # always take ZAMS binaries to step_detached
         if event == 'ZAMS' and state in BINARY_STATES_ZAMS: # check for event
             MY_FLOW_CHART[key] = 'step_detached'
+    
+    ## Add new step to binary step list
+    BINARY_STEPS_ALL.append('step_HMS_HMS_RLO')
```

Comment:
That's the wrong place. This function is not called at the beginning of the simulation, but whenever the flow chart is requested.
Maybe you can keep it here, but you should then add a check, that you only add the entry when it's not existing already otherwise this can become a memory leek.

Reply:
Ok, you will see in the changes that I ended-up getting the step names from the binary SimulationProperties attribute and creating a list of the names in the Simulation Properties class, rather than the flow.